### PR TITLE
build: switch to use system octomap

### LINKED ISSUE #482: rolling: octomap rosdep not found
https://github.com/ANYbotics/grid_map/actions/runs/11736778137/job/32696522337?pr=481#step:5:51

This blocks CI working, so we can't merge any changes. I'm quite time constrained, anyone is welcome to propose a fix.

### PR DESCRIPTION
**Pull request by wep21**
- closes #482  
- related to https://github.com/ros/rosdistro/issues/41622

### CODE REVIEW

**Comment by wep21** [comment]
```
 /opt/ros/rolling/include/octomap_msgs/conversions.h 
```
```
/opt/ros/rolling/include/sensor_msgs/sensor_msgs/image_encodings.hpp
```
octomap_msgs conversions.h seems to be placed in the wrong directory.

**Comment by Ryanf55** [comment]
> ```
>  /opt/ros/rolling/include/octomap_msgs/conversions.h 
> ```
> 
> ```
> /opt/ros/rolling/include/sensor_msgs/sensor_msgs/image_encodings.hpp
> ```
> 
> octomap_msgs conversions.h seems to be placed in the wrong directory.

Fixed by this PR: https://github.com/OctoMap/octomap_msgs/pull/20

I requested them to do a release. In the mean time, I think this is an ok patch to get our CI running again.

**Review by Ryanf55 — APPROVED** [review]

**Comment by Ryanf55** [comment]
The octomap maintainers have issues PR's to ROSDIstro. I expect those to go in a few days, and this to be available shortly. Perhaps we wait a bit before merging.

**Comment by kjalloul-anybotics** [comment]
Hey do you think this fix can be merged now? The [issue they linked](https://github.com/OctoMap/octomap_msgs/issues/22) was updated and closed.

File: grid_map_demos/CMakeLists.txt
Comment (wep21) [inline grid_map_demos/CMakeLists.txt:150]:
```suggestion
```

**Pull request merged**
